lib/service/src/main/java/io/atlasmap/service/AtlasService.java

```diff
@@ -347,343 +230,138 @@ public Response resetAllMappings() {
+    public Response uploadLibrary(InputStream requestIn) {
+        if (requestIn == null) {
+            throw new WebApplicationException("No library file found in request body");
         }
-    }
-
-    /**
-     * @deprecated use {@link #updateMappingRequest(InputStream, Integer, UriInfo)} instead.
-     * @param mapping mapping
-     * @param uriInfo URI info
-     * @return .
-     */
-    @Deprecated
-    @POST
-    @Path("/mapping")
-    @Consumes(MediaType.APPLICATION_JSON)
-    @Produces(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Update Mapping", description = "Update existing mapping file on the server")
-    @RequestBody(description = "Mapping file content", content = @Content(schema = @Schema(implementation = AtlasMapping.class)))
-    @ApiResponses(@ApiResponse(responseCode = "200", description = "Succeeded"))
-    public Response updateMappingRequestOld(
-            InputStream mapping,
-            @Context UriInfo uriInfo)
-    {
-        return updateMappingRequest(mapping, 0, uriInfo);
-    }
 
-    /**
-     * Updates existing mapping file on the server.
-     * @param mapping mapping
-     * @param mappingDefinitionId mapping definition ID
-     * @param uriInfo URI info
-     * @return empty response
-     */
-    @POST
-    @Path("/mapping/{mappingDefinitionId}")
-    @Consumes(MediaType.APPLICATION_JSON)
-    @Produces(MediaType.APPLICATION_JSON)
-    @Operation(summary = "Update Mapping", description = "Update existing mapping file on the server")
-    @RequestBody(description = "Mapping file content", content = @Content(schema = @Schema(implementation = AtlasMapping.class)))
-    @ApiResponses(@ApiResponse(responseCode = "200", description = "Succeeded"))
-    public Response updateMappingRequest(
-            InputStream mapping,
-            @Parameter(description = "Mapping Definition ID") @PathParam("mappingDefinitionId") Integer mappingDefinitionId,
-            @Context UriInfo uriInfo) {
-        ADMArchiveHandler handler = loadExplodedMappingDirectory(mappingDefinitionId);
-        UriBuilder builder = uriInfo.getAbsolutePathBuilder();
         try {
-            handler.setMappingDefinitionBytes(mapping);
-            handler.persist();
-            builder.path(handler.getMappingDefinition().getName());
-        } catch (AtlasException e) {
-            LOG.error("Error saving Mapping Definition file.\n" + e.getMessage(), e);
-            throw new WebApplicationException(e.getMessage(), e, Status.INTERNAL_SERVER_ERROR);
+            getLibraryLoader().addJarFromStream(requestIn);
+        } catch (Exception e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.error("", e);
+            }
+            StringBuilder buf = new StringBuilder();
+            buf.append("Failed to import a jar file. This error occurs when:\n")
+                    .append(("\t1. The jar file is not compatible with the JVM AtlasMap backend server is running on\n"))
```

Comment:
The jar file is not compatible with the JVM AtlasMap backend server is running on is not a correct sentence.  I'd just shorted in to:
The jar file is not compatible with the JVM AtlasMap backend server

Reply:
added "which", `the JVM which AtlasMap backend server is running on`